modules/highgui/src/window.cpp

```diff
@@ -662,7 +662,7 @@ int cv::waitKey(int delay)
     static int use_legacy = -1;
     if (use_legacy < 0)
     {
-        use_legacy = getenv("OPENCV_LEGACY_WAITKEY") != NULL ? 1 : 0;
```

Comment:
I propose to drop it in 5.x

Reply:
OK, but this PR is for 4.x, so we'll need to remove it after these changes make their way to 5.x.